src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -734,6 +773,16 @@ export class ConnectionWidget extends lifecycle.Disposable {
 				this._tableContainer.classList.add('hide-azure-accounts');
 			}
 
+			if (this._customOptionWidgets) {
+				for (let i = 0; i < this._customOptionWidgets.length; i++) {
+					if (this._customOptionWidgets[i] instanceof SelectBox) {
+						(this._customOptionWidgets[i] as SelectBox).selectWithOptionName(this.getModelValue(connectionInfo.options[this._customOptions[i].name]));
```

Comment:
If you use foreach you can simplify this and get type narrowing

(I would also suggest initializing it to an empty array so you don't have to have the null check)

```typescript
this._customOptionWidgets.forEach(widget => {
    if (widget instanceof SelectBox) {
        widget.selectWithOptionName(...)
    ...
}
```

Reply:
Can't use for that one as I need index to get `name` from `_customOptions`, but I've used it in other places where possible.

Reply:
```typescript
this._customOptionWidgets.forEach( (widget, i) => {

}
```